src/nvim/lua/stdlib.c

```diff
@@ -538,4 +573,8 @@ void nlua_state_add_stdlib(lua_State *const lstate, bool is_thread)
   // vim.json
   lua_cjson_new(lstate);
   lua_setfield(lstate, -2, "json");
+
+  // vim.iconv
+  lua_pushcfunction(lstate, &nlua_iconv);
+  lua_setfield(lstate, -2, "iconv");
```

Comment:
`iconv_string` accesses `p_ambw` and `p_emoji` internally. Therefore, it is not thread-safe. I do not agree with exposing vim.iconv to threads as it is currently implemented, although the cases where it becomes a problem may be rare.

Reply:
I'm not sure what the code in question is intended to do, but I don't think it makes sense to replace characters that cannot be converted with `vim.iconv()` with the number of cell widths "`?`".